eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/DefaultEventStoreTransaction.java

```diff
@@ -77,15 +79,34 @@ public DefaultEventStoreTransaction(@Nonnull AsyncEventStorageEngine eventStorag
     }
 
     @Override
-    public MessageStream<? extends EventMessage<?>> source(@Nonnull SourcingCondition condition,
-                                                           @Nonnull ProcessingContext context) {
-        context.updateResource(
+    public MessageStream<? extends EventMessage<?>> source(@Nonnull SourcingCondition condition) {
+        var appendCondition = processingContext.updateResource(
                 appendConditionKey,
-                appendCondition -> appendCondition == null
+                ac -> ac == null
                         ? AppendCondition.withCriteria(condition.criteria())
-                        : appendCondition.orCriteria(condition.criteria())
+                        : ac.orCriteria(condition.criteria())
         );
-        return eventStorageEngine.source(condition);
+        MessageStream<EventMessage<?>> source = eventStorageEngine.source(condition);
+        if (appendCondition.consistencyMarker() == ConsistencyMarker.ORIGIN) {
```

Comment:
This appendCondition will (probably) only be `ORIGIN` on the first `source` invocation. 

Let's say we source two models, A and B:

- source model A
- append condition is ORIGIN, so we wrap the MessageStream and use the highest loaded marker as appendCondition, saving this in the context. Let's say it's globalIndex marker is 1000
- Now we source model B. The appendCondition has a marker of 1000, so this if is never invoked, and the stream is never wrapped


However, let's say that model B's consistency marker would have been 800, this would lead to 200 events in the store that are not checked for appending, right?